org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XSISchemaModel.java

```diff
@@ -43,55 +78,42 @@ public static void computeCompletionResponses(ICompletionRequest request,
+			documentation = TYPE_DOC;
 			name = actualPrefix + ":type";
-			createCompletionItem(name, isSnippetsSupported, snippet, editRange, response, documentation);
+			createCompletionItem(name, isSnippetsSupported, snippet, editRange, response, documentation, generateValue);
 		}
 		//The xsi:schemaLocation and xsi:noNamespaceSchemaLocation attributes can be used in a document 
 		//to provide hints as to the physical location of schema documents which may be used for ·assessment·.
-		if(!schemaLocationExists && !noNamespaceSchemaLocationExists) {
-			documentation = "The xsi:schemaLocation attribute can be used in an XML document " +
-			"to reference an XML Schema document that has a target namespace.\r\n  " + 
-			"```xml  \r\n" +
-			"<ns:root xmlns:ns=\"http://example.com/ns\"" +
-            " xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"" +
-            " xsi:schemaLocation=\"http://example.com/ns example-ns.xsd\">\r\n " +
-            "  <!-- ... -->\r\n  " +
-            "</ns:root>\r\n  " +
-			"```" ;
+		if(inRootElement && !schemaLocationExists && !noNamespaceSchemaLocationExists) {
+			documentation = NO_NAMESPACE_SCHEMA_LOCATION_DOC;
 			name = actualPrefix + ":schemaLocation";
-			createCompletionItem(name, isSnippetsSupported, snippet, editRange, response, documentation);
+			createCompletionItem(name, isSnippetsSupported, snippet, editRange, response, documentation, generateValue);
 			
-			documentation = "The xsi:noNamespaceSchemaLocation attribute can be used in an XML document " +
-			"to reference an XML Schema document that does not have a target namespace.\r\n  " + 
-			"```xml  \r\n" +
-			"\r\n<root xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"  "+
-			"  xsi:noNamespaceSchemaLocation=\"example.xsd\">  \r\n" +
-			"  <!-- ... -->  \r\n" +
-			"</root>  \r\n" +
-			"```" ;
+			documentation = SCHEMA_LOCATION_DOC;
 			name = actualPrefix + ":noNamespaceSchemaLocation";
-			createCompletionItem(name, isSnippetsSupported, snippet, editRange, response, documentation);		}
+			createCompletionItem(name, isSnippetsSupported, snippet, editRange, response, documentation, generateValue);		
+		}
 	}
 
 	private static void createCompletionItem(String text, boolean isSnippetsSupported, 
-			String defaultAndSnippet, Range editRange, ICompletionResponse response, String documentation) {
+			String defaultAndSnippet, Range editRange, ICompletionResponse response, String documentation, boolean generateValue) {
 		CompletionItem item = new CompletionItem();
 		item.setLabel(text);
 		item.setKind(CompletionItemKind.Value);
 		item.setFilterText(text);
 		item.setInsertTextFormat(isSnippetsSupported ? InsertTextFormat.Snippet : InsertTextFormat.PlainText);
-		item.setTextEdit(new TextEdit(editRange, text + "=\"" + defaultAndSnippet + "\""));
+		text = generateValue ? text + "=\"" + defaultAndSnippet + "\"" : text + "$0";
```

Comment:
$0 placeholder can only be used if snippets are supported

Reply:
Please note I will push a new class AttributeCompletionItem will simplify the creation of completion item for attribute (takes care of snippet support, generate choice snippet if attributes has values (like "true", "false")